### PR TITLE
feat: Hedgey Integration, adding locked_token_distributions table and…

### PR DESCRIPTION
**Pull request by bark4mark**
## Motivation and Context

We require a new table (locked_token_distributions) and the ability to update an existing circle_integration.

## Description

This is the backend changes for the Hedgey distribution

## Test and Deployment Plan

A developer should apply the migrations and ensure they don't fail, there is no functionality affected by this change.

## Reviewers

@levity @crabsinger

### CODE REVIEW

File: hasura/metadata/databases/default/tables/public_locked_token_distributions.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+insert_permissions:
+  - role: user
+    permission:
+      check:
+        epoch:
+          circle:
+            users:
+              profile:
+                id:
+                  _eq: X-Hasura-User-Id
```
Comment (levity) [inline hasura/metadata/databases/default/tables/public_locked_token_distributions.yaml:17]: The insert and update checks should also validate that the user is a circle admin (role = 1)

File: hasura/migrations/default/1670234780227_create_table_public_locked_token_distributions/up.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+CREATE TABLE "public"."locked_token_distributions" ("id" bigserial NOT NULL, "epoch_id" bigint NOT NULL, "gift_amount" numeric NOT NULL, "tx_hash" varchar, "distribution_json" jsonb NOT NULL, "distributed_by" varchar NOT NULL, PRIMARY KEY ("id") , FOREIGN KEY ("epoch_id") REFERENCES "public"."epoches"("id") ON UPDATE cascade ON DELETE cascade);
```
Comment (levity) [inline hasura/migrations/default/1670234780227_create_table_public_locked_token_distributions/up.sql:1]: What values does `distributed_by` contain?

**Comment by levity** [comment]
Thanks for making these small PRs!

File: hasura/migrations/default/1670234780227_create_table_public_locked_token_distributions/up.sql (hunk 2 above)
Comment (bark4mark) [reply hasura/migrations/default/1670234780227_create_table_public_locked_token_distributions/up.sql:1]: This is the address that kicked off the distribution, we will display it in the locked token distribution table.

**Comment by bark4mark** [comment]
I have answered the two questions, thank you :grin:

File: hasura/migrations/default/1670234780227_create_table_public_locked_token_distributions/up.sql (hunk 2 above)
Comment (levity) [reply hasura/migrations/default/1670234780227_create_table_public_locked_token_distributions/up.sql:1]: could we make it into a foreign key linking to `profiles.id`? or do you want the flexibility of being able to add an address that doesn't correspond to a Coordinape user?

**Review by levity — COMMENTED** [review]
Followup question: https://github.com/coordinape/coordinape/pull/1733#discussion_r1042734490

File: hasura/migrations/default/1670234780227_create_table_public_locked_token_distributions/up.sql (hunk 2 above)
Comment (bark4mark) [reply hasura/migrations/default/1670234780227_create_table_public_locked_token_distributions/up.sql]: @levity yes that makes sense, I've updated the database to have a foreign key and added the rule to have it populated by the session variable (similar to how the distributions table is configured).

**Comment by levity** [comment]
I'm going to merge this to staging first to make sure the migrations take; if all goes well, we can push to prod tomorrow

**Pull request merged**